integration-tests/utils/integration-helpers.js

```diff
@@ -72,9 +75,94 @@ const withUser = test => async args => {
   }
 }
 
+const addUserToBilling = withUserToken(async ({ email, token }) => {
+  const ajaxPage = await context.newPage()
+
+  await ajaxPage.goto(testUrl)
+  await signIntoTerra(ajaxPage, token)
+
+  await ajaxPage.evaluate((email, billingProject) => {
+    return window.Ajax().Billing.project(billingProject).addUser(['User'], email)
+  }, email, billingProject)
+  await ajaxPage.close()
+})
+
+const removeUserFromBilling = withUserToken(async ({ email, token }) => {
+  const ajaxPage = await context.newPage()
+
+  await ajaxPage.goto(testUrl)
+  await signIntoTerra(ajaxPage, token)
+
+  await ajaxPage.evaluate((email, billingProject) => {
+    return window.Ajax().Billing.project(billingProject).removeUser(['User'], email)
+  }, email, billingProject)
+  await ajaxPage.close()
+})
+
+const withBilling = test => async options => {
+  await addUserToBilling(options)
+
+  try {
+    await test({ ...options })
+  } finally {
+    await deleteCluster(options)
+    await removeUserFromBilling(options)
+  }
+}
+
+const trimClustersOldestFirst = _.flow(
+  _.remove({ status: 'Deleting' }),
+  _.sortBy('createdDate')
+)
+
+const currentCluster = _.flow(trimClustersOldestFirst, _.last)
+
+const getCurrentCluster = withUserToken(async ({ email, token }) => {
+  const ajaxPage = await context.newPage()
+  await ajaxPage.goto(testUrl)
+  await signIntoTerra(ajaxPage, token)
+
+  const clusters = await ajaxPage.evaluate((email, billingProject) => {
+    return window.Ajax().Clusters.list({ googleProject: billingProject })
+  }, billingProject, email)
+
+  await ajaxPage.close()
+  return currentCluster(clusters)
+})
+
+const deleteCluster = withUserToken(async ({ email, token }) => {
+  const ajaxPage = await context.newPage()
+  await ajaxPage.goto(testUrl)
+  await signIntoTerra(ajaxPage, token)
+
+  const currentC = await getCurrentCluster()
+  await ajaxPage.evaluate((currentC, email, billingProject) => {
+    return window.Ajax().Clusters.cluster(billingProject, currentC.clusterName).delete()
+  }, currentC, email, billingProject)
+
+  await ajaxPage.close()
+})
+
+const withRegisteredUser = test => withUser(async ({ page, token, ...args }) => {
+  const ajaxPage = await context.newPage()
+
+  await ajaxPage.goto(testUrl)
+  await click(ajaxPage, clickable({ textContains: 'View Workspaces' }))
+  await signIntoTerra(ajaxPage, token)
+  await dismissNotifications(ajaxPage)
+  await fillIn(ajaxPage, input({ labelContains: 'First Name' }), 'Integration')
+  await fillIn(ajaxPage, input({ labelContains: 'Last Name' }), 'Test')
+  await click(ajaxPage, clickable({ textContains: 'Register' }))
+  await click(ajaxPage, clickable({ textContains: 'Accept' }))
+  await ajaxPage.close()
```

Comment:
It's a little surprising that we click 'Accept' both here and also in the test cases. I suspect what might be happening is that the click happens, but the page is closed before the network request completes, so the TOS isn't actually accepted here. If so, we should an additional wait for some piece of page content to ensure that the TOS acceptance is recorded before we tear down the page.

Reply:
Thank you for pointing that out - I should have called that out as it seemed wrong but I could not determine how to fix it. I will try making that adjustment and see if it resolves the need to double accept.